presto-spark-base/src/main/java/com/facebook/presto/spark/planner/optimizers/PickJoinSides.java

```diff
@@ -104,7 +104,8 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
         // if we don't have exact costs for the join, but based on source tables we think the left side
         // is very small or much smaller than the right, then flip the join.
         if (rightSize > leftSize || (isSizeBasedJoinDistributionTypeEnabled(context.getSession()) && (Double.isNaN(leftSize) || Double.isNaN(rightSize)) && isLeftSideSmall(joinNode, context))) {
-            rewrittenNode = createRuntimeSwappedJoinNode(joinNode, metadata, sqlParser, context.getLookup(), context.getSession(), context.getVariableAllocator(), context.getIdAllocator());
+            // This is never used for Prestissimo.
```

Comment:
i don't like that this assumption isn't enforced by the code.  Could we instead pass the native execution config to this optimizer like we do with other optimizers (may need to inject FeaturesConfig into adaptivePlanOptimizers, but that's fine).